### PR TITLE
[WIP] New package: Cockatrice-2.9.0

### PR DESCRIPTION
**Pull request by jason1987d**
#### Testing the changes
- I tested the changes in this PR: **YES**|**briefly**|**NO**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

### CODE REVIEW

**Comment by jason1987d** [comment]
Given this is the error, any suggestions on how to fix?

```
/usr/bin/ld: common/libcockatrice_common.a(debug_pb_message.cpp.o): undefined reference to symbol '_ZN4absl12lts_2023080213base_internal12SpinLockWaitEPSt6atomicIjEiPKNS1_22SpinLockWaitTransitionENS1_14SchedulingModeE'
/usr/bin/ld: /usr/lib/libabsl_spinlock_wait.so.2308.0.0: error adding symbols: DSO missing from command line
```

**Comment by robertek** [comment]
You may try to clean-up your dependencies. You mix up a lot of things together.

hostmakedepends: tools to build the package (you definitely do not need there the qt6-xxx stuff)
makedepends: devel packages for required libraries, the pkg-config is extra there
depends: the packages it should install on the system, without the ones that are resolved automatic (library depends).

**Comment by jason1987d** [comment]
I tried changing those protobuf dependencies, which uses version 24 by default, to protobuf28 (and devel too), it got much further, but it looks like another args or something may be required to handle this new type of error?

```
=> Cockatrice-2.9.0_1: running pre-pkg hook: 04-generate-runtime-deps ...
   SONAME: libQt6Multimedia.so.6 <-> qt6-multimedia>=6.3.1_1
   SONAME: libQt6PrintSupport.so.6 <-> qt6-printsupport>=6.0.0_1
   SONAME: libQt6WebSockets.so.6 <-> qt6-websockets>=6.2.0alpha_1
   SONAME: libQt6Widgets.so.6 <-> qt6-widgets>=6.0.0_1
   SONAME: libprotobuf.so.28 <-> UNKNOWN PKG PLEASE FIX!
   SONAME: libQt6Gui.so.6 <-> qt6-gui>=6.0.0_1
   SONAME: libQt6Network.so.6 <-> qt6-network>=6.0.0_1
   SONAME: libQt6Core.so.6 <-> qt6-core>=6.0.0_1
   SONAME: libstdc++.so.6 <-> libstdc++>=4.4.0_1
   SONAME: libm.so.6 <-> glibc>=2.36_1
   SONAME: libgcc_s.so.1 <-> libgcc>=4.4.0_1
   SONAME: libc.so.6 <-> glibc>=2.36_1
   SONAME: ld-linux-x86-64.so.2 <-> glibc>=2.36_1
   SONAME: libQt6Sql.so.6 <-> qt6-sql>=6.0.0_1
   SONAME: libz.so.1 <-> zlib>=1.2.3_1
   SONAME: liblzma.so.5 <-> liblzma>=5.0.0_1
=> ERROR: Cockatrice-2.9.0_1: cannot guess required shlibs, aborting!
```

File: srcpkgs/Cockatrice/template [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+version=2.9.0
+revision=1
+build_style=cmake
+configure_args="-DWITH_SERVER=1"
+hostmakedepends="cmake git liblzma pkg-config extra-cmake-modules protobuf zlib"
+makedepends="qt6-base-devel liblzma-devel protobuf-devel qt6-svg-devel qt6-tools-devel qt6-websockets-devel qt6-multimedia-devel abseil-cpp-devel"
+depends="qt6-core qt6-base qt6-sql mariadb qt6-tools zlib libprotobuf"
+short_desc="Cross-platform virtual tabletop for multiplayer card games"
+maintainer="Jason Elswick <jason@jasondavid.tv>"
+license="GPL-3.0-or-later"
```
Comment (Luciogi) [inline srcpkgs/Cockatrice/template:12]:
```suggestion
license="GPL-2.0-only"
```

**Comment by jason1987d** [comment]
With @Luciogi here there has also been some discussion about updating protobuf, which not only has confusing versioning, is orphaned, and is a major version behind (that I know of, 25.1). Out of curiosity I attempted to update the template for this version, even setting noverifyrdeps=yes and this was the result.

```
=> libprotobuf-lite-25.1_1: running pre-pkg hook: 99-pkglint ...
=> ERROR: libprotobuf-lite-25.1_1: SONAME bump detected: libprotobuf-lite.so.24.1.0 -> libprotobuf-lite.so.25.1.0
=> ERROR: libprotobuf-lite-25.1_1: please update common/shlibs with this line: "libprotobuf-lite.so.25.1.0 libprotobuf-lite-25.1_1"
=> ERROR: libprotobuf-lite-25.1_1: all reverse dependencies should also be revbumped to be rebuilt against libprotobuf-lite.so.25.1.0:
=> ERROR:    EternalTerminal-6.2.1_4
=> ERROR:    libvlc-3.0.20_1
=> ERROR:    mixxx-2.3.3_6
=> ERROR:    qgis-3.34.1_1
=> ERROR:    telegram-desktop-4.11.2_1
=> ERROR: libprotobuf-lite-25.1_1: cannot continue with installation!
=> ERROR: libprotobuf-lite-25.1_1: pre-pkg_99-pkglint: 'grep -E "${_pattern}" $mapshlibs' exited with 1
=> ERROR:   in hook() at common/hooks/pre-pkg/99-pkglint.sh:173
=> ERROR:   in run_func() at common/xbps-src/shutils/common.sh:57
=> ERROR:   in run_pkg_hooks() at common/xbps-src/shutils/common.sh:298
=> ERROR:   in main() at common/xbps-src/libexec/xbps-src-prepkg.sh:47
```

**Comment by Luciogi** [comment]
actually updating `protobuf` is not fixing linking error

**Comment by Luciogi** [comment]
as protobuf25 old package (which is deprecated) works, my guess is that current package `protobuf` has different build option which is causing abnormal linking here

**Pull request closed without merging**